client/control/components/data/QueryAPIToken.js

```diff
@@ -0,0 +1,38 @@
+import { Component, PropTypes as pt } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { requestAPIToken } from '../../actions/authActions.js';
+
+class QueryAPIToken extends Component {
+  static propTypes = {
+    tokenFetching: pt.bool,
+    token: pt.object,
+    requestAPIToken: pt.func.isRequired,
+  }
+
+  componentWillMount() {
+    this.request();
+  }
+
+  request() {
+    if (!this.props.tokenFetching && !this.props.token) {
```

Comment:
Are there any cases of the token/Fetching to change after the app has been running for a bit? More specifically, do we need a `componentWillReceiveProps` function that compares `newProps.token` and `this.props.token` or something?